Client/Frontend/Contextual Hint/ContextualHintViewModel.swift

```diff
@@ -30,6 +30,7 @@ class ContextualHintViewModel: ContextualHintPrefsKeysProvider {
     private var profile: Profile
     private var hasSentTelemetryEvent = false
     var arrowDirection = UIPopoverArrowDirection.down
+    var overlayState: OverlayStateProtocol?
```

Comment:
`ContextualHintViewModel` could get new tests. Probably just testing we react as we should with and without `overlayState` defined. (For example, I see overlayState is nil from WallpaperManager).

Reply:
Added `test_shouldPresentInactiveTabsHint_WithNilOverlayMode()` at `ContextualHintEligibilityUtility`